### PR TITLE
 BUG: make Series handle dtype='int64' for string array

### LINKED ISSUE #44923: BUG: using dtype='int64' argument of Series causes `ValueError: values cannot be losslessly cast to int64` for integer strings
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the master branch of pandas.

### Reproducible Example

```python
import pandas as pd
pd.Series(['1', '2'], dtype='int64')
```

### Issue Description

```
C:\Users\a\AppData\Local\Programs\Python\Python310\lib\site-packages\numpy\core\numeric.py:2446: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  return bool(asarray(a1 == a2).all())
Traceback (most recent call last):
  File "C:\Users\a\AppData\Roaming\JetBrains\PyCharmCE2021.3\scratches\scratch_3.py", line 2, in <module>
    pd.Series(['1', '2'], dtype='int64')
  File "f:\prog\pandas\pandas\core\series.py", line 450, in __init__
    data = sanitize_array(data, index, dtype, copy)
  File "f:\prog\pandas\pandas\core\construction.py", line 583, in sanitize_array
    subarr = _try_cast(data, dtype, copy, raise_cast_failure)
  File "f:\prog\pandas\pandas\core\construction.py", line 768, in _try_cast
    subarr = maybe_cast_to_integer_array(arr, dtype)
  File "f:\prog\pandas\pandas\core\dtypes\cast.py", line 2100, in maybe_cast_to_integer_array
    raise ValueError(f"values cannot be losslessly cast to {dtype}")
ValueError: values cannot be losslessly cast to int64
```

### Expected Behavior

I expect 
`pd.Series(['1', '2'], dtype='int64')` 
to work fine and return a Series object with dtype int64, just like how 
`pd.Series(['1', '2'], dtype='float64')` 
returns a float64 Series. 

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 39ccb352b53e9b5b694c4f7f044774f9c3677e98
python           : 3.10.1.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.19043
machine          : AMD64
processor        : Intel64 Family 6 Model 94 Stepping 3, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : English_United States.1252
pandas           : 0+untagged.1.g39ccb35
numpy            : 1.21.4
pytz             : 2021.3
dateutil         : 2.8.2
pip              : 21.3.1
setuptools       : 59.5.0
Cython           : 0.29.25
pytest           : 6.2.5
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : 3.0.2
lxml.etree       : 4.6.3
html5lib         : 1.1
pymysql          : None
psycopg2         : None
jinja2           : 3.0.3
IPython          : 7.29.0
pandas_datareader: None
bs4              : 4.10.0
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : 3.5.0
numexpr          : None
odfpy            : None
openpyxl         : 3.0.9
pandas_gbq       : None
pyarrow          : None
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by 5j9**
- [x] closes #44923
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by 5j9** [comment]
breaks some tests, e.g. `pd.Series([1, 2, 3.5], dtype=int)`.

**Review by phofl — CHANGES_REQUESTED** [review]
I think this disables to many checks, for example:

```
x = pd.Series(['300', '2'], dtype='int8')
```

This one finishes without warning and converts 300 to 44
while 
```
x = pd.Series([300, 2], dtype='int8')
```

produces a FutureWarning and will raise in the future. I think we should check, if we are causing overflows here too

File: pandas/tests/series/test_constructors.py [hunk 1]
```diff
@@ -1880,6 +1880,18 @@ def test_constructor_bool_dtype_missing_values(self):
...
         tm.assert_series_equal(result, expected)
 
+    def test_constructor_int64_dtype(self, any_int_dtype):
+        # GH#44923
+        result = Series(["0", "1", "2"], dtype=any_int_dtype)
+        expected = Series([0, 1, 2], dtype=any_int_dtype)
+        tm.assert_series_equal(result, expected)
+
+    def test_constructor_raise_on_lossy_conversion_of_strings(self):
+        with pytest.raises(
```
Comment (phofl) [inline pandas/tests/series/test_constructors.py:1890]: Can you add gh ref here too and a whatsnew?

Comment (5j9) [reply pandas/tests/series/test_constructors.py:1891]: done

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
Please move whatsnew to 1.6.0

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @5j9

**Pull request merged**